urld558a3d988_gao715108023_resourceMonitor_tgz-pJ8-com_sinopec_agent_AgentJ8/src/com/sinopec/agent/infocollect/DynamicMemInfo.java

```diff
@@ -0,0 +1,124 @@
+ */
+public class DynamicMemInfo implements Runnable {
+
+	private static final Log LOG = LogFactory.getLog(DynamicMemInfo.class);
+
+	private int samplingInterval;
+
+	private volatile boolean running = true;
+
+	public DynamicMemInfo(int samplingInterval) {
+		this.samplingInterval = samplingInterval;
+	}
+
+	public void getMemInfoFromProc() {
+		int totalMem = 0;
+		int freeMem = 0;
+		// int totalSwap = 0;
+		// int freeSwap = 0;
+		// int usedSwap = 0;
+		int usedMem = 0;
+		FileReader reader = null;
+		BufferedReader br = null;
+		try {
+			reader = new FileReader("/proc/meminfo");
+			br = new BufferedReader(reader);
+			String str = null;
+
+			while ((str = br.readLine()) != null) {
+				StringBuilder sbb = new StringBuilder();
+				char c = ' ';
+				for (int i = 0; i < str.length(); i++) {
+					char ch = str.charAt(i);
+					if (ch != c) {
+						sbb.append(ch);
+					}
+				}
+				str = sbb.toString();
+				String a[] = str.split(":");
+				if (a[0].equals("MemTotal")) {
+					Matcher m = Pattern.compile("\\d+").matcher(a[1]);
+					while (m.find()) {
+						totalMem = totalMem + Integer.parseInt(m.group(0));
+						totalMem = totalMem / 1024;
+					}
+				}
+				if (a[0].equals("MemFree")) {
+					Matcher m = Pattern.compile("\\d+").matcher(a[1]);
+					while (m.find()) {
+						freeMem = freeMem + Integer.parseInt(m.group(0));
+						freeMem = freeMem / 1024;
+					}
+				}
+				// if (a[0].equals("SwapTotal")) {
+				// Matcher m = Pattern.compile("\\d+").matcher(a[1]);
+				// while (m.find()) {
+				// totalSwap = totalSwap + Integer.parseInt(m.group(0));
+				// totalSwap = totalSwap / 1024;
+				// }
+				// }
+				// if (a[0].equals("SwapFree")) {
+				// Matcher m = Pattern.compile("\\d+").matcher(a[1]);
+				// while (m.find()) {
+				// freeSwap = freeSwap + Integer.parseInt(m.group(0));
+				// freeSwap = freeSwap / 1024;
+				// }
+				// }
+				// usedSwap = totalSwap - freeSwap;
+				usedMem = totalMem - freeMem;
+			}
+			SharedArea.dynamicInfoSerializable.setTotalMemory(totalMem);
+			SharedArea.dynamicInfoSerializable.setUsedMemory(usedMem);
+			SharedArea.dynamicInfoSerializable.setIdleMemory(freeMem);
+			// dynamicInfoSerializable.setSwapSize(totalSwap);
+			// dynamicInfoSerializable.setUsedSwap(usedSwap);
+			// dynamicInfoSerializable.setIdleSwap(freeSwap);
+		} catch (FileNotFoundException e) {
+			LOG.error(LogUtils.getTrace(e));
+		} catch (IOException e) {
+			LOG.error(LogUtils.getTrace(e));
+		} finally {
+			try {
+				if (br != null)
+					br.close();
+				if (reader != null)
+					reader.close();
+			} catch (IOException e) {
+				// TODO Auto-generated catch block
+				LOG.error(LogUtils.getTrace(e));
+			}
+		}
+	}
+
+	@Override
+	public void run() {
+		// To change body of implemented methods use File | Settings | File
+		// Templates.
+		while (running) {
+			getMemInfoFromProc();
+			try {
+				Thread.sleep(samplingInterval);
+			} catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)